activerecord/lib/active_record/persistence.rb

```diff
@@ -67,8 +67,14 @@ def create!(attributes = nil, &block)
       # how this "single-table" inheritance mapping is implemented.
       def instantiate(attributes, column_types = {}, &block)
         klass = discriminate_class_for_record(attributes)
+        instantiate_instance_of(klass, attributes, column_types, &block)
+      end
+
+      # Given a class, an attributes hash, +instantiate+ returns a new instance
+      # of the class. Accepts only keys as strings.
+      def instantiate_instance_of(klass, attributes, column_types = {}, &block)
```

Comment:
Do we want/expect people to call this method on their applications? If not this should be `# :nodoc:`

Reply:
I don't think we want people to call this.  I'll `:nodoc:` it.

Reply:
👍